gpm_api/tests/test_dataset/test_granule.py

```diff
@@ -299,26 +293,40 @@ def construct_dataset_and_check_validation(input_datetimes, expected_datetimes):
 
 def test_finalize_dataset(monkeypatch):
```

Comment:
It's a very long test function. Can't be break down in a couple of smaller test units? 
 `test_finalize_dataset_crs`
 `test_finalize_dataset_reshaping`
 `test_finalize_dataset_time_subsetting`
 `test_finalize_dataset_time_encoding`
 `test_finalize_dataset_attrs`